anchore_engine/services/policy_engine/engine/feeds/config.py

```diff
@@ -0,0 +1,284 @@
+                    provider,
+                )
+                continue
+
+    else:
+        logger.debug(
+            "Configuration found for data feeds not found, falling back to %s provider defaults",
+            provider,
+        )
+        to_be_synced = {
+            feed_name: sync_config
+            for feed_name, sync_config in DEFAULT_PROVIDER_SYNC_CONFIG_MAP.get(
+                provider
+            ).items()
+            if sync_config.enabled
+        }
+
+    logger.info(
+        "Data feeds to be synced for %s provider: %s",
+        provider,
+        list(to_be_synced.keys()),
+    )
+
+    return to_be_synced
+
+
+def is_sync_enabled(vulnerabilities_config: Dict) -> bool:
+    """
+    Returns whether a feeds sync is enabled based on the provider config. Returns true if config sections are missing
+    to be backwards compatible
+
+    :param vulnerabilities_config: vulnerabilities section of config file
+    {
+       "provider": "legacy",
+       "sync": {
+          "enabled": True,
+          "ssl_verify": True,
+          "connection_timeout_seconds": 3,
+          "read_timeout_seconds": 60,
+          "data": {
+             "vulnerabilities": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             },
+             "nvdv2": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             },
+             "github": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             }
+          }
+       }
+    }
+    :return: True or False
+    """
+    if not vulnerabilities_config:
+        return True
+
+    sync_enabled = vulnerabilities_config.get("sync", {}).get("enabled")
+    if isinstance(sync_enabled, bool):
+        return sync_enabled
+    else:
+        return True
+
+
+def get_provider(vulnerabilities_config: Dict) -> str:
+    """
+    Returns the provider if it's configured or a default provider
+
+    :param vulnerabilities_config: vulnerabilities section of config file
+    {
+       "provider": "legacy",
+       "sync": {
+          "enabled": True,
+          "ssl_verify": True,
+          "connection_timeout_seconds": 3,
+          "read_timeout_seconds": 60,
+          "data": {
+             "vulnerabilities": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             },
+             "nvdv2": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             },
+             "github": {
+                "enabled": True,
+                "url": "https://ancho.re/v1/service/feeds"
+             }
+          }
+       }
+    }
+    """
+    if not vulnerabilities_config:
+        return DEFAULT_VULNERABILITIES_PROVIDER
+
+    try:
+        provider = SupportedProvider(vulnerabilities_config.get("provider"))
+        provider = provider.value
```

Comment:
i feel like this defeats the purpose of using an enum. String comparisons are brittle, so you should return the enum and check against the enum in the caller.

Reply:
The purpose of using an enum here is 
1. to keep a collection of normalized providers in one place and avoid sprinkling literal strings through out the system. When we add a new provider, that enum should be updated along 
2. to validate the supplied provider

 I understand this doesn't have to be an enum necessarily but defining a constant for each provider and then create a list with all of them was additional steps that didn't seem necessary

Reply:
Yep, i'm saying you should return the enum, not the value, and compare against the enum in the caller.